pkgs/development/tools/godot/3/default.nix

```diff
@@ -1,108 +1,138 @@
-{ stdenv
-, lib
+{ stdenv, lib
+, alsa-lib
+, alsa-plugins
 , fetchFromGitHub
-, scons
-, pkg-config
-, udev
+, freetype
+, installShellFiles
+, libGLU
+, libpulseaudio
 , libX11
 , libXcursor
+, libXext
+, libXfixes
+, libXi
 , libXinerama
 , libXrandr
 , libXrender
-, libpulseaudio
-, libXi
-, libXext
-, libXfixes
-, freetype
-, openssl
-, alsa-lib
-, alsa-plugins
 , makeWrapper
-, libGLU
-, zlib
+, openssl
+, pkg-config
+, scons
+, udev
 , yasm
-, withUdev ? true
+, zlib
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self: {
   pname = "godot";
   version = "3.5.1";
 
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
+    rev = "${self.version}-stable";
```

Comment:
this should probably use `finalAttrs`

Reply:
There's multiple naming conventions for these things.  One is `oldAttrs` and `finalAttrs`, the other is `super` and `self`.  As far as I can tell, one is not predominantly more common than the other, and I am used to the latter.

Reply:
okay fine